### PR TITLE
navbar dropdown

### LINKED ISSUE #35: Implement dropdown menu in navbar with Stimulus
Adapt existing html template and design to feature
Create new Stimulus controller for click event
Use Stimulus data-action, data-controller and data-controller-target to handle event in html

### PR DESCRIPTION
**Pull request by MarianaCarlota**
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,11 +5,18 @@
     "@hotwired/stimulus": "^3.2.1",
     "@hotwired/turbo-rails": "^7.2.5",
     "@popperjs/core": "^2.11.6",
+    "autoprefixer": "^9",
```
Comment (wJoenn) [inline package.json]: Can you remove this line

File: package.json [hunk 2]
```diff
@@ -5,11 +5,18 @@
...
     "bootstrap": "^5.2.3",
+    "el-transition": "^0.0.7",
+    "postcss": "^7",
+    "tailwindcss": "npm:@tailwindcss/postcss7-compat",
```
Comment (wJoenn) [inline package.json]: these 3 lines

File: package.json [hunk 3]
```diff
@@ -5,11 +5,18 @@
...
     "webpack": "^5.75.0",
     "webpack-cli": "^5.0.1"
   },
   "scripts": {
     "build": "webpack --config webpack.config.js"
+  },
+  "devDependencies": {
+    "dropdown-menu-basic": "^1.2.0"
   }
```
Comment (wJoenn) [inline package.json]: And these 3 lines please ?

File: app/views/shared/_navbar.html.erb [hunk 4]
```diff
@@ -11,7 +11,24 @@
...
-  <div class="dropdown">
-    <%= image_tag "avatar.png", id: "dropdown" %>
+  <%# Dropdown avatar %>
+  <div class="dropdown" data-controller="dropdown-menu">
+    <button type="button" data-dropdown-menu-target="button" data-action="click->dropdown-menu#toggleMenu" class="btn-drop" aria-haspopup="true" aria-expanded="true">
+    </button>
+
+    <div class="menu-options hidden" data-dropdown-menu-target="menu">
+      <div>
+        <div class="options" role="menu" aria-orientation="vertical">
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb]: Can you remove the aria-orientation dataset ? You don't need it anymore

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -11,7 +11,24 @@
...
+          <%= link_to "See all pieces of art", arts_path, role: "menuitem", class: "option" %>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb]: You don't need the role anymore too so it can be removed

File: app/views/shared/_navbar.html.erb [hunk 6]
```diff
@@ -11,7 +11,24 @@
...
+          <button type="submit" class="btn-gold" role="menuitem">
+            <% if current_user %>
+              <%= link_to "log out", destroy_user_session_path %>
+            <% else %>
+              <%= link_to "log in", user_session_path %>
+            <% end %>
+          </button>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb]: There's no reason to have a link inside a button
Remove the button element and just keep the if else with the links inside.
The you can give the btn-default class and the btn-gold class to both link_to
to log_out use turbo-method: :delete in the link_to

File: app/javascript/controllers/dropdown_menu_controller.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { Controller } from "@hotwired/stimulus";
+import {enter, leave, toggle} from 'el-transition';
```
Comment (wJoenn) [inline app/javascript/controllers/dropdown_menu_controller.js]: You can delete this line

File: app/javascript/controllers/dropdown_menu_controller.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+console.log("hello from dropdown controller");
```
Comment (wJoenn) [inline app/javascript/controllers/dropdown_menu_controller.js]: You can delete this too

File: app/javascript/controllers/dropdown_menu_controller.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+export default class extends Controller {
+  static targets = ["menu", "button"]
+
+  connect() {
+    // call the enter and leave functions
+  }
```
Comment (wJoenn) [inline app/javascript/controllers/dropdown_menu_controller.js]: We're not using the connect method so let's get rid of it for now

File: app/javascript/controllers/dropdown_menu_controller.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+  toggleMenu() {
+    if(this.menuTarget.classList.contains('hidden')) {
+        enter(this.menuTarget)
+    } else {
+        leave(this.menuTarget)
+    }
+}
```
Comment (wJoenn) [inline app/javascript/controllers/dropdown_menu_controller.js]: You can remove the if else statement and just toggle the class "hidden" on click

File: app/assets/stylesheets/config/_colors.scss [hunk 11]
```diff
@@ -7,9 +7,10 @@ $yellow: #FFC65A;
...
 $green: #1EDD88;
 $gray: #0E0000;
-$light-gray: #F4F4F4;
+$light-gray: #f4f4f4;
 
 //custom colors:
 $main-grey: #181a1b;
 $secondary-grey: #222425;
 $gold: rgb(236, 196, 134);
+$lightgrey: #d3d3d3;
```
Comment (wJoenn) [inline app/assets/stylesheets/config/_colors.scss]: This can be removed

File: app/assets/stylesheets/components/_navbar.scss [hunk 12]
```diff
@@ -70,11 +70,45 @@
...
+  transition: max-height 0.3s ease;
+  position: absolute;
+  top: calc(100% + 10px);
+  right: 0;
+  z-index: 10;
+}
+
+.options {
+  text-align: left;
+  width: 95%;
```
Comment (wJoenn) [inline app/assets/stylesheets/components/_navbar.scss]: padding would make more sense

**Review by wJoenn — APPROVED** [review]

**Review by Glod33 — APPROVED** [review]

**Review by wJoenn — CHANGES_REQUESTED** [review]
Review planned for tonight

**Review by wJoenn — APPROVED** [review]

**Pull request merged**